src/imap/mod.rs

```diff
@@ -769,6 +827,39 @@ impl Imap {
         Ok(new_msgs)
     }
 
+    /// Like fetch_after(), but not for new messages but existing ones (the DC_FETCH_EXISTING_MSGS_COUNT newest messages)
+    async fn fetch_existing_msgs_prefetch(
+        &mut self,
+    ) -> Result<BTreeMap<u32, async_imap::types::Fetch>> {
+        let exists: i32 = {
+            let mailbox = self.config.selected_mailbox.as_ref();
+            let mailbox = mailbox.context("fetch_existing_msgs_prefetch(): no mailbox selected")?;
+            mailbox.exists.try_into().context("exists was too large")?
```

Comment:
Is it always about too large number?

Reply:
Yes, `try_into()` is needed to convert u32 to i32 and this conversion can only fail if the number is too large for an i32. It needs to be signed so that the `exists - DC_FETCH_EXISTING_MSGS_COUNT` doesn't panic if exists is smaller than DC_FETCH_EXISTING_MSGS_COUNT.

Anyway, I just made it an i64 so that we can use `into().`